umbraco/Umbraco-CMS
# Migrates more cookie auth and IWebSecurity

* IWebSecurity is now migrated and simplified
* Cookie authentication is now actually working
* The session id claim handling is now entirely only part of the cookie auth flow, not our back office identity - since it is specific to cookies only
* GetRemainingAuthSeconds is migrated and is done using a custom claim added during the auth process (which is a much better approach)
* The installation step is not responsible for logging the user in, it's now just done at the end of the install sequence in a controller
* AuthenticationController.GetCurrentUser() migrated (there's a TODO there for which a task has been created)
* ConfigureBackOfficeCookieOptions does most of the cookie auth heavy lifting, much of this logic was scattered in various places in v8:
  * Ensures the current User is set in the current request on login
  * Validates the security stamp
  * Validates the user session id
  * Ensures the right claims are applied on each request
  * Ensures the right cookies are all cleared on logout
* Ensures all ajax requests from the back office have the header X-Requested-With which is required by the cookie auth handler in order to deal with authorization results and whether to redirect or return http status codes
* Converts FromClaimsIdentity to a boolean method to avoid try/catch

## Testing

* rebuild the gulp dev task for the back office ui
* Ensure you can login 
* Open developer tools and look at xhr requests on the network tab
* Refresh the back office - you will remain logged in (i.e. you won't be redirect back to the login screen)
* Have a look at the network tab, you'll see requests for GetCurrentUser and the request headers will contain X-Requested-With header
